src/Runtime/ExecutionSession.cpp

```diff
@@ -79,42 +65,32 @@ const std::string *ExecutionSession::queryEntryPoints(
 void ExecutionSession::setEntryPoint(const std::string &entryPointName) {
   _entryPointFunc = reinterpret_cast<entryPointFuncType>(
       _sharedLibraryHandle.getAddressOfSymbol(entryPointName.c_str()));
-  if (!_entryPointFunc) {
-    std::stringstream errStr;
-    errStr << "Cannot load symbol: '" << entryPointName << "'" << std::endl;
-    throw std::runtime_error(errStr.str());
-  }
+  if (!_entryPointFunc)
+    throw std::runtime_error(reportSymbolLoadingError(entryPointName));
   _entryPointName = entryPointName;
+  errno = 0; // No errors.
 }
 
 std::vector<OMTensorUniquePtr> ExecutionSession::run(
     std::vector<OMTensorUniquePtr> ins) {
-  if (!_entryPointFunc) {
-    std::stringstream errStr;
-    errStr << "Must set the entry point before calling run function"
-           << std::endl;
-    throw std::runtime_error(errStr.str());
-  }
+  if (!_entryPointFunc)
+    throw std::runtime_error(reportMissingEntryPointError("run"));
```

Comment:
This seems misleading to me - when I see `reportMissingEntryPointError("run")`, I would assume that the missing entry point is `run`, not that there has to be an entry point set before calling `run`.

Reply:
Same, probably you want to report that "run_main_graph"  (whatever the default entry point name is) was not present.

Reply:
Changed the name to `reportUndefinedEntryPointIn("run")`. Hopefully that reads better.